typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,129 @@
+import * as bindingsJsGrpc from '../../kurtosis_core_rpc_api_bindings/api_container_service_grpc_pb'; //TODO - should I be importing all modules or just specific ones for best practices (need to check)
+import * as bindingsJs from '../../kurtosis_core_rpc_api_bindings/api_container_service_pb'; //TODO (comment) - I cannont merge with above line unfortunately, can only import one module at a time MAXIMUM
+import { ServiceID } from './service';
+import { StaticFileID } from './container_creation_config'; 
+import * as grpc from "grpc"; //TODO - this is needed for grpc callback, but this import is tentative 
+import * as constructors from "../constructor_calls";
+var path = require("path"); //TODO - I don't think this works
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+class GeneratedFileFilepaths { //TODO (comment) - try interface if class doesn't work
+	absoluteFilepathOnTestsuiteContainer: string;
+	absoluteFilepathOnServiceContainer:   string;
+}
+
+class ServiceContext {
+    
+    client: bindingsJsGrpc.ApiContainerServiceClient;
```

Comment:
These fields should be `private readonly` by default

Reply:
Ah right, encapsulation is important! But hold on, taking a look at the documentation, wouldn't it be better to change this to `readonly` instead of `private readonly` since `readonly` prevents assignment to fields outside of the constructor (https://www.typescriptlang.org/docs/handbook/2/classes.html#readonly) while `private` prevents visibility to anyone even to the subclasses (https://www.typescriptlang.org/docs/handbook/2/classes.html#private). Of course, this means we can't do inheritance. Would that be okay?